app/controllers/admin/schedules_controller.rb

```diff
@@ -57,17 +77,38 @@ def adapt_params
       params[:schedule][:order_cycle_ids] = params[:order_cycle_ids]
     end
 
-    def check_editable_order_cycle_ids
+    def check_editable_order_cycle_ids_create
+      return unless params[:order_cycle_ids]
+
+      requested = params[:order_cycle_ids]
+      @existing_order_cycle_ids = []
+
+      result = editable_order_cycles(requested)
+
+      params[:order_cycle_ids] = result
+    end
+
+    def check_editable_order_cycle_ids_update
       return unless params[:schedule][:order_cycle_ids]
 
       requested = params[:schedule][:order_cycle_ids]
-      @existing_order_cycle_ids = @schedule.persisted? ? @schedule.order_cycle_ids : []
-      permitted = OrderCycle.where(id: params[:schedule][:order_cycle_ids] | @existing_order_cycle_ids).merge(OrderCycle.managed_by(spree_current_user)).pluck(:id)
+      @existing_order_cycle_ids = @schedule.order_cycle_ids
+
+      result = editable_order_cycles(requested)
+
+      params[:schedule][:order_cycle_ids] = result
+      @schedule.order_cycle_ids = result
+    end
```

Comment:
I'd maybe change the method name and remove local variable assignments, they dont add much do they?

```ruby
    def define_editable_order_cycle_ids_for_update
      return unless params[:schedule][:order_cycle_ids]

      @existing_order_cycle_ids = @schedule.order_cycle_ids
      params[:schedule][:order_cycle_ids] = editable_order_cycles(params[:schedule][:order_cycle_ids])

      @schedule.order_cycle_ids = result
    end
```

Reply:
yes, good catch